### PR TITLE
CVE-2011-3884 AND CVE-2014-1727

### PR DESCRIPTION
**Pull request by Zach-DiPasquale**
CVE-2011-3884 and CVE-2014-1727
Zachary DiPasquale
RIT SWEN-331

ToDo:
Part 1 complete and passing:
- [x] CVE-2011-3884
- [x] CVE-2014-1727

Part 2 complete and passing:
- [x] CVE-2011-3884
- [x] CVE-2014-1727

### CODE REVIEW

**Comment by Zach-DiPasquale** [comment]
Part 1 complete

**Comment by LithiumLeech** [comment]
CWE 399 is a category. Consider picking CWE 401, 416, or maybe another more appropriate one.

File: cves/CVE-2011-3884.yml [hunk 1]
```diff
@@ -4,14 +4,14 @@ CWE_instructions: |
   Please go to cwe.mitre.org and find the most specific, appropriate CWE entry
   that describes your vulnerability. (Tip: this may not be a good one to start
   with - spend time understanding this vulnerability before making your choice!)
-CWE: 
+CWE: 399
```
Comment (LithiumLeech) [inline cves/CVE-2011-3884.yml]: CWE 399 is a category. Consider picking CWE 401, 416, or maybe another more appropriate one.

File: cves/CVE-2011-3884.yml [hunk 2]
```diff
@@ -32,7 +32,17 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: |
+  A timing issues in the DOM traversal can cause denial of service attacks and
```
Comment (LithiumLeech) [inline cves/CVE-2011-3884.yml]: I think it would helpful to add an explanation of DOM

File: cves/CVE-2011-3884.yml [hunk 3]
```diff
@@ -32,7 +32,17 @@ description_instructions: |
...
+  remote vector issues or possible unknown issuess through a crafted document.
+
+  The DOM traversal runs for at most 50ms before pausing to allow the WebkKit 
+  thread to run. If the frame is removed before the traversal resumes then it
+  will attempt to access a deleted WebFrame.
+
+  The fix is to ensure that the WebDocument is not null.
+
+  Interesting enough these vulnerabilities are in a client side phishing detection model.
```
Comment (LithiumLeech) [inline cves/CVE-2011-3884.yml]: I think something along the lines of "An interesting note is that..." would be better. But, at least add a comma after enough.

File: cves/CVE-2011-3884.yml [hunk 4]
```diff
@@ -52,8 +62,19 @@ fixes_vcc_instructions: |
   CVE-2011-3092.yml). Fixes and VCCs follow the same format.
 fixes:
 - :commit: bab0878f91a962788069c3c2d059be1acb18a4e6
-  :note: ''
-vccs: []
+  :note: |
+    Since the is a time limit on how long the detector can run 
+    it is not guaranteed that the WebFrame will exist the fix was
+    to cache the WebDocument pointer instead of the WebFrame and
```
Comment (LithiumLeech) [inline cves/CVE-2011-3884.yml]: missing some punctuation. WebFrame will exist. The fix was...

File: cves/CVE-2011-3884.yml [hunk 5]
```diff
@@ -74,9 +95,13 @@ unit_tested:
 
     For the "fix" answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
-  answer: 
-  code: 
-  fix: 
+  answer: |
+    There are tests in this part of the code, but the discovery of this bug was
+    not found through unit test.
```
Comment (LithiumLeech) [inline cves/CVE-2011-3884.yml]: unit testing

File: cves/CVE-2011-3884.yml [hunk 6]
```diff
@@ -192,4 +227,11 @@ mistakes:
...
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer: 
+  answer: |
+    The developers did not consider all the consequences for adding a time out.
+    It appears they were attempting to make the process quicker, since they even
+    reduced the limit one time. 
+
+    Managing resources are an important part of security. Since you can not allways
+    sure that a resource or object will be availible. Check for edge cases and attempt
```
Comment (LithiumLeech) [inline cves/CVE-2011-3884.yml]: *is an important* *always be* *object is available*

File: cves/CVE-2011-3884.yml [hunk 7]
```diff
@@ -160,8 +193,10 @@ lessons:
     applies: 
     note: 
   distrust_input:
-    applies: 
-    note: 
+    applies: true
+    note: |
+      They didn't consider if an object would(n't) be available ater considering
+      there is a max run time.
```
Comment (LithiumLeech) [inline cves/CVE-2011-3884.yml]: I didn't study this cve as much as you, but I'm not sure I understand why you chose distrust_input. Maybe defensive coding would be a better category?

File: cves/CVE-2014-1727.yml [hunk 8]
```diff
@@ -4,14 +4,14 @@ CWE_instructions: |
   Please go to cwe.mitre.org and find the most specific, appropriate CWE entry
   that describes your vulnerability. (Tip: this may not be a good one to start
   with - spend time understanding this vulnerability before making your choice!)
-CWE: 
+CWE: 399
```
Comment (LithiumLeech) [inline cves/CVE-2014-1727.yml]: Consider CWE 416

File: cves/CVE-2014-1727.yml [hunk 9]
```diff
@@ -32,7 +32,10 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: |
+  A vulnerability in the a web color chooser UI. The object is removed in the wrong
+  class allowing an attacker to access to freed memory. This can cause denial of 
+  service attacks, or possible other attacks on unknown vectors related to forms.
```
Comment (LithiumLeech) [inline cves/CVE-2014-1727.yml]: --the a web color
--attacker to access to freed

File: cves/CVE-2014-1727.yml [hunk 10]
```diff
@@ -54,8 +57,12 @@ fixes_vcc_instructions: |
   CVE-2011-3092.yml). Fixes and VCCs follow the same format.
 fixes:
 - :commit: c0be143193b370e1e1ee268e480f15f2684e1d66
-  :note: ''
-vccs: []
+  :note: |
+    Override the destruct method so that when RenderViewImpl is destroyed
+    it doesnt also destroy RendererWebColorChooserImpl.
```
Comment (LithiumLeech) [inline cves/CVE-2014-1727.yml]: not sure if it's Overrided or Overrode. But, past tense would be better here.

File: cves/CVE-2014-1727.yml [hunk 11]
```diff
@@ -150,8 +161,11 @@ lessons:
     If you think of another lesson we covered in class that applies here, feel
     free to give it a small name and add one in the same format as these.
   defense_in_depth:
-    applies: 
-    note: 
+    applies: true
+    note: |
+      Precautions were not take to limit the possiblity of another object from
+      deleting it. If there was some defense mechanism in place this could have
+      been prevented.
```
Comment (LithiumLeech) [inline cves/CVE-2014-1727.yml]: *teken^

File: cves/CVE-2014-1727.yml [hunk 12]
```diff
@@ -194,4 +208,14 @@ mistakes:
...
+    This vulnerability is related to resource management. We should take into 
+    account when each paticular object may be deleted and ensure that it is
+    ok to be deleted at that time.
+
+    This is something very hard to protect against. There are many different times
+    an object can be deleted and figuring out all posible interactions can be tough.
+
+    Its best to limit the possiblities of object iteractions and ensure that it only
+    does what it needs to do and when. It was an esay fix for a bug that could have big
+    implications since its a use-after-free vulnerability.
```
Comment (LithiumLeech) [inline cves/CVE-2014-1727.yml]: *easy fix^ *have had big^ *it's^

File: cves/CVE-2011-3884.yml (hunk 2 above)
Comment (Zach-DiPasquale) [reply cves/CVE-2011-3884.yml]: Will do!

File: cves/CVE-2011-3884.yml (hunk 7 above)
Comment (Zach-DiPasquale) [reply cves/CVE-2011-3884.yml]: Yeah definitely. I'll move it there.

File: cves/CVE-2014-1727.yml (hunk 9 above)
Comment (Zach-DiPasquale) [reply cves/CVE-2014-1727.yml]: 🤦‍♂️

**Pull request merged**